controllers/openstackdeploy_controller.go

```diff
@@ -164,14 +164,22 @@ func (r *OpenStackDeployReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	}
 
 	advancedSettings := instance.Spec.AdvancedSettings.DeepCopy()
-	if advancedSettings.Playbook == "" {
+	if len(advancedSettings.Playbooks) == 0 {
 		switch instance.Spec.Mode {
 		case "update":
-			advancedSettings.Playbook = "update_steps_playbook.yaml"
+			advancedSettings.Playbooks = []string{"update_steps_playbook.yaml"}
+		case "upgrade":
+			advancedSettings.Playbooks = []string{
```

Comment:
isn't  "upgrade-prepare.yaml" supposed to be here ? 

https://github.com/openstack/tripleo-heat-templates/blob/master/environments/lifecycle/upgrade-prepare.yaml#L12

Reply:
No, the equivalent is the configgenerator